src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/StructBuilder.java

```diff
@@ -48,14 +48,18 @@
     private final GroupLayout structLayout;
     private final Type structType;
     private final Deque<String> prefixElementNames;
-    private final String safeParameterName;
+    private final String setterParameterName;
 
     StructBuilder(JavaSourceBuilder enclosing, String className, GroupLayout structLayout, Type structType) {
         super(enclosing, Kind.CLASS, className);
         this.structLayout = structLayout;
         this.structType = structType;
         prefixElementNames = new ArrayDeque<>();
-        safeParameterName = isEnclosedBySameName("x")? "x$" : "x";
+        setterParameterName = safeParameterName("x");
```

Comment:
Can we avoid the caching and just call `safeParameterName` where required? I think caching like this makes code generation more brittle than it has to, as it introduces an implicit dependency. In other words, in a way it is an "accident" that all methods generated in struct builder accept a parameter named "x" - I'd like not to rely on this assumption too much, and make sure that _all_ parameters of _all_ methods in any of the nested builders (StructBuilder, FunctionalInterfaceBuilder) are validated accordingly.

For instance, in `emitIndexedFieldSetter` the problem is not just `x` - the problem is also the `seg` and the `index` parameter (as the struct could be called `seg` or `index`). In other words, I think the parameter name should be mangled in a lot more places.